pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -129,30 +145,49 @@ stdenv.mkDerivation {
   ];
 
   cmakeFlags = [
-    "-DGPU_TARGET=${gpuTargetString}"
-    (lib.cmakeBool "MAGMA_ENABLE_CUDA" cudaSupport)
-    (lib.cmakeBool "MAGMA_ENABLE_HIP" rocmSupport)
-  ] ++ lists.optionals static [
-    "-DBUILD_SHARED_LIBS=OFF"
+    (strings.cmakeFeature "GPU_TARGET" gpuTargetString)
+    (strings.cmakeBool "MAGMA_ENABLE_CUDA" cudaSupport)
+    (strings.cmakeBool "MAGMA_ENABLE_HIP" rocmSupport)
+    (strings.cmakeBool "BUILD_SHARED_LIBS" (!static))
+    # Set the Fortran name mangling scheme explicitly. We must set FORTRAN_CONVENTION manually because it will
+    # otherwise not be set in NVCC_FLAGS or DEVCCFLAGS (which we cannot modify).
+    # See https://github.com/NixOS/nixpkgs/issues/281656#issuecomment-1902931289
+    (strings.cmakeBool "USE_FORTRAN" true)
+    (strings.cmakeFeature "CMAKE_C_FLAGS" "-DADD_")
+    (strings.cmakeFeature "CMAKE_CXX_FLAGS" "-DADD_")
+    (strings.cmakeFeature "FORTRAN_CONVENTION" "-DADD_")
   ] ++ lists.optionals cudaSupport [
-    "-DCMAKE_CUDA_ARCHITECTURES=${cudaArchitecturesString}"
-    "-DMIN_ARCH=${minArch}" # Disarms magma's asserts
-    "-DCMAKE_C_COMPILER=${backendStdenv.cc}/bin/cc"
-    "-DCMAKE_CXX_COMPILER=${backendStdenv.cc}/bin/c++"
+    (strings.cmakeFeature "CMAKE_CUDA_ARCHITECTURES" cudaArchitecturesString)
+    (strings.cmakeFeature "MIN_ARCH" minArch) # Disarms magma's asserts
   ] ++ lists.optionals rocmSupport [
-    "-DCMAKE_C_COMPILER=${rocmPackages.clr}/bin/hipcc"
-    "-DCMAKE_CXX_COMPILER=${rocmPackages.clr}/bin/hipcc"
-  ] ++ lists.optionals (cudaPackages.cudaAtLeast "12.0.0") [
-    (lib.cmakeBool "USE_FORTRAN" false)
-  ];
-
-  buildFlags = [
-    "magma"
-    "magma_sparse"
+    (strings.cmakeFeature "CMAKE_C_COMPILER" "${rocmPackages.clr}/bin/hipcc")
+    (strings.cmakeFeature "CMAKE_CXX_COMPILER" "${rocmPackages.clr}/bin/hipcc")
   ];
 
+  # Magma doesn't have a test suite we can easily run, just loose executables.
+  # NOTE: The python script `run_tests.py` requires you to be in the same directory as the test
+  # executables. That means `cd`'ing into `$test/bin/` before running it.
```

Comment:
Couldn't we just cd into the directory before running it? Perhaps I'm missing something?

Reply:
Your comment is what I had hoped to communicate in the note. The `run_tests.py` script is essentially the entry point for Magma's test suite, but it looks for the test executables in the directory which contains the script.

What I had meant by the note was that running `./result-test/bin/run_tests.py` won't work; one needs to `cd ./result-test/bin && ./run_tests.py`.

Does that make sense? How can I better word that to get it across?

Reply:
Ah that's because they require GPU acceleration and I don't remember if we've resolved how to best modify the sandbox during the `checkPhase` to allow that. That's also why the test executables are all put in a `test` output -- they're meant to be run outside of the main build where a GPU is available.

Does that answer your questions?

Reply:
Ah ok so the matter of how `run_tests.py` is configured is besides the point then? No way to run GPU tests in `checkPhase` for now anyways.

Reply:
It is; I've removed the comment because it's really about how to run the test suite once the derivation is built, not really about *how* the derivation is built.

Reply:
Gotcha gotcha, ok thanks for helping explain!